### DISCUSSION TITLE
[BlenderBIM] How to change IfcOpenShell v0.7 to v0.8 in BlenderBIM?

### DISCUSSION THREAD

**Discussion by Martin15135215** [IfcOpenShell/IfcOpenShell#4795] [Q&A] [answered]
So i have:
- Blender version 4.1 (Python 3.11)
- BlenderBIM v0.0.240602 (blenderbim-240602-py311-win.zip)
- Windows 10

And then i got the IfcOpenShell v0.8 for windows Python 3.11 from here:

https://anaconda.org/ifcopenshell/ifcopenshell/files

The file named: win-64_ifcopenshell-0.8.0alpha1-py311h9a73951_1.tar.bz2 -> In there, there is a ifcopenshell folder under:
\Lib\site-packages\

Then i just swapped the ifcopenshell folder unter this path:

C:\Users\{YOUR_USER}\AppData\Roaming\Blender Foundation\X.XX\scripts\addons\blenderbim\libs\site\packages

Unfortunately, that did not exactly work out...

The primary reason i wanted IfcOpenShell v0.8 is, that i can view the following ifc entities:
 
 [IfcAlignment](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcAlignment.htm)
[IfcAlignmentCant](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcAlignmentCant.htm)
 [IfcAlignmentHorizontal](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcAlignmentHorizontal.htm)
 [IfcAlignmentSegment](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcAlignmentSegment.htm)
 [IfcAlignmentVertical](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcAlignmentVertical.htm)

**Comment by RickBrice** [comment, marked as answer]
I have successfully got BlenderBIM to work with IfcOpenShell v0.8.0 on one computer. I've tried on a second computer, and it doesn't work, though they are seemingly exactly the same. In a nutshell, this is what I did

- Build IFCOS v0.8.0 from source
- Install Blender and BlenderBIM add-on per the normal instructions https://docs.blenderbim.org/users/installation.html
- Followed the "Live development environment" installation instructions https://docs.blenderbim.org/devs/installation.html

I have been able to open models that use IfcAlignment (and related) with geometric representations using IfcCompositeCurve, IfcGradientCurve, IfcSegmentedReferenceCurve, and IfcCurveSegment as well as bridge element geometry positioned with IfcLinearPlacement and represented with IfcSectionedSolidHorizontal. However, the geometric representations of the alignment curves does not display in BlenderBIM (see https://github.com/IfcOpenShell/IfcOpenShell/issues/4727).

Perhaps that's helpful, perhaps not.

**Reply by Martin15135215** [reply]
Thanks for the reply! I will look into it tomorrow.

**Reply by Martin15135215** [reply]
Hi @RickBrice 

Thanks for pointing me to the issue! If there are at the moment no visualizations in BlenderBIM for Alignment entities possible, I wait till the issue is resolved before I try installing v0.8 in BlenderBIM. I subscribed to that issue.

I also saw [here](https://community.osarch.org/discussion/comment/20822/#Comment_20822), that BlenderBIM will change the IfcOpenShell from v0.7 to v0.8 in " a few days or so ".  Therefore, I think the issue of installing BlenderBIM with IfcOpenShell v0.8 will be made possible with the unstable builds shortly.
